pyteal/compiler/scratchslots.py

```diff
@@ -107,6 +108,7 @@ def assignScratchSlotsToSubroutines(
                 "Slot ID {} has been assigned multiple times".format(slot.id)
             )
         slotIds.add(slot.id)
+        reserved_ids.add(slot.id)
```

Comment:
@algoidurovic Optional - I spent a few min refactoring here because I got tripped up on the usage of `continue`.  Here's a version that passes all tests with the added benefit of being able to list _all_ duplicate slot IDs (though I didn't update the error message).

```
    def find_reserved_slot_ids(slots: typing.Set[ScratchSlot]) -> typing.Set[int]:
        ids = [slot.id for slot in slots if slot.isReservedSlot]
        duplicates = [k for k, xs in itertools.groupby(sorted(ids)) if len(list(xs)) > 1]

        if len(duplicates) > 0:
            raise TealInternalError("Duplicate IDs assigned")

        return set(ids)

    reserved_ids = find_reserved_slot_ids(allSlots)
    slotIds = reserved_ids.copy()
```

Requires a few imports:

```
import typing
import itertools
```

If you feel it's worth taking in, you're welcome to use it.  Otherwise let's resolve the thread.

Reply:
Resolving because my recent commit changed this part of the code and I don't think the feedback is applicable anymore.